libindy/src/services/anoncreds/prover.rs

```diff
@@ -413,13 +436,23 @@ impl Prover {
 
     pub fn extend_proof_request_restrictions(&self,
                                              version: &ProofRequestsVersion,
-                                             name: &str,
+                                             name: &Option<String>,
+                                             names: &Option<Vec<String>>,
                                              referent: &str,
                                              restrictions: &Option<Query>,
                                              extra_query: &Option<&ProofRequestExtraQuery>) -> IndyResult<Query> {
-        let mut queries: Vec<Query> = vec![
-            Query::Eq(format!("attr::{}::marker", &attr_common_view(name)), ATTRIBUTE_EXISTENCE_MARKER.to_string())
-        ];
+        info!("name: {:?}, names: {:?}", name, names);
+        let mut queries: Vec<Query> = if let Some(name) = name {
```

Comment:
Have we considered the option to merge `name` and `names` parameters for this function? It may be more straightforward.

Reply:
Good catch, found a cute way to do it